app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -79,6 +79,32 @@ def metrics_connect
                                      authentication_password('default'))
     end
 
+    def run_generic_operation(operation, path)
```

Comment:
yeah, I kind of agree...we have something similar on the vmware side and it's nasty (see `invoke_vim_ws`).  The better pattern we've set up is to call with_provider_object or with_provider_connection, but that might be tough in this case...it seems the operations API is not designed in an OO manner like that.